src/Version/CMakeLists.txt

```diff
@@ -53,4 +53,24 @@ set_source_files_properties("${version_inc}"
              HEADER_FILE_ONLY TRUE
   )
 
-add_custom_target(OMVersion ALL DEPENDS ${version_inc})
+if (NOT DEFINED ONNX_MLIR_VERSION)
+  set(ONNX_MLIR_VERSION "0.3.0")
+endif()
+list(APPEND DEFINITIONS "ONNX_MLIR_VERSION=\"${ONNX_MLIR_VERSION}\"")
+if (NOT DEFINED ONNX_VERSION)
+  set(ONNX_VERSION "1.11.0")
+endif()
+list(APPEND DEFINITIONS "ONNX_VERSION=\"${ONNX_VERSION}\"")
+if (ONNX_MLIR_VENDOR)
+  list(APPEND DEFINITIONS "ONNX_MLIR_VENDOR=\"${ONNX_MLIR_VENDOR }\"")
+endif()
+list(APPEND DEFINITIONS "LLVM_PACKAGE_VERSION=\"${LLVM_PACKAGE_VERSION}\"")
+set_property(SOURCE Version.cpp APPEND PROPERTY COMPILE_DEFINITIONS ${DEFINITIONS})
```

Comment:
Why not use `target_compile_definitions`?

Reply:
Changed to use `target_compile_definitions`.